src/btree/bt_sync_obsolete.c

```diff
@@ -94,11 +105,28 @@ __sync_obsolete_inmem_evict(WT_SESSION_IMPL *session, WT_REF *ref)
         /* Mark the obsolete page to evict soon. */
         __wt_page_evict_soon(session, ref);
         WT_STAT_CONN_DSRC_INCR(session, checkpoint_cleanup_pages_evict);
+    } else if (S2BT(session)->cc_obsolete_tw_pages <
+        S2C(session)->heuristic_controls.cc_obsolete_tw_pages_dirty &&
+      !WT_TIME_AGGREGATE_HAS_STOP(&newest_ta)) {
+        newest_ts = WT_MAX(newest_ta.newest_start_durable_ts, newest_ta.newest_stop_durable_ts);
+        if ((newest_ta.newest_txn != WT_TXN_NONE || newest_ts != WT_TS_NONE) &&
+          __wt_txn_visible_all(session, newest_ta.newest_txn, newest_ts)) {
+            /*
+             * Dirty the page with an obsolete time window to let the page reconciliation remove all
+             * the obsolete time window information.
+             */
+            __wt_verbose(session, WT_VERB_CHECKPOINT_CLEANUP,
+              "%p in-memory page %s obsolete time window: time aggregate %s", (void *)ref, tag,
+              __wt_time_aggregate_to_string(&newest_ta, time_string));
+
+            WT_RET(__wt_page_modify_init(session, ref->page));
+            __wt_page_modify_set(session, ref->page);
+
+            S2BT(session)->cc_obsolete_tw_pages++;
```

Comment:
nit: use ++S2BT(session)->cc_obsolete_tw_pages?

Reply:
The compiler should be able to optimise `++i` vs `i++` automatically.